src/Features/Core/Portable/CodeFixesAndRefactorings/CodeActionRequestPriorityProvider.MutableProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeActions
+{
+    internal abstract partial class CodeActionRequestPriorityProvider
+    {
+        private sealed class MutableProvider : CodeActionRequestPriorityProvider
+        {
+            private readonly object _gate;
+
+            /// <summary>
+            /// Set of analyzers which have been de-prioritized to <see cref="CodeActionRequestPriority.Low"/> bucket.
+            /// </summary>
+            private ImmutableHashSet<DiagnosticAnalyzer> _lowPriorityAnalyzers = ImmutableHashSet<DiagnosticAnalyzer>.Empty;
+
+            private MutableProvider(CodeActionRequestPriority priority, ImmutableHashSet<DiagnosticAnalyzer> lowPriorityAnalyzers)
+                : base(priority)
+            {
+                _lowPriorityAnalyzers = lowPriorityAnalyzers;
```

Comment:
oh interesting.  this can be instantiated with an immutable set.  ok.  i need to think more on this.